caf-logging-serilog/Sanitizer.cs

```diff
@@ -0,0 +1,135 @@
+        {
+            if (processId is ScalarValue && ThreadId is ScalarValue)
+            {
+                var threadId = ThreadId.ToString();
+                return new ScalarValue("#" + processId.ToString().Substring(0, 3) + "." + threadId.PadLeft(3, '0'));
+            }
+
+            return null;
+        }
+
+        public static LogEventPropertyValue? Sanitize(LogEventPropertyValue? value, LogEventPropertyValue length, LogEventPropertyValue padding)
+        {
+
+            if (value is ScalarValue)
+            {
+                int valueLength, valuePadding;
+                try
+                {
+                    valueLength = Convert.ToInt32(length.ToString());
+                    valuePadding = Convert.ToInt32(padding.ToString());
+                }
+                catch (FormatException)
+                {
+                    valueLength = 4;
+                    valuePadding = 4;
+                }
+                var newValue = rgx.Replace(value.ToString(), "");
+                if (valueLength <= newValue.Length)
+                {
+                    newValue = newValue.Substring(0, valueLength);
+                }
+                newValue = newValue.PadRight(valuePadding, ' ');
+                return new ScalarValue(newValue);
+            }
+
+            return null;
+        }
+
+        public static LogEventPropertyValue? MaybeJsonMsgAndEx(LogEventPropertyValue? messageValue, LogEventPropertyValue? exceptionValue)
+        {
+
+            if (messageValue is ScalarValue)
+            {
+                var message = messageValue.ToString();
+                if (IsMessageSafeToLog(message) && null == exceptionValue)
+                {
+                    return new ScalarValue(message);
+                }
+                else 
+                {
+                    if (null != exceptionValue)
+                    {
+                        var exception = exceptionValue.ToString();
+                        return new ScalarValue(JsonConvert.SerializeObject(new SanitizedMessage(message, exception)));
+                    } 
+                    else
+                    {
+                        return new ScalarValue(JsonConvert.SerializeObject(new SanitizedMessage(message)));
+                    }
+                }
+            }
+            return null;
+        }
+
+        private static bool IsMessageSafeToLog(string message)
+        {
+            if (string.IsNullOrWhiteSpace(message))
+            {
+                return true;
+            }
+
+            if (message.StartsWith("{") || message.StartsWith("\"{"))
+            {
+                return false;
+            }
+
+            // Makes sure that each character is ASCII printable
+            // http://www.csc.villanova.edu/~tway/resources/ascii-table.html
+            foreach (char c in message)
+            {
+                if (char.IsControl(c))
```

Comment:
This check doesn't relate to the comment above.  Are we checking if the characters are ASCII, or are we checking if they are control characters?